Cython/Utility/TypeConversion.c

```diff
@@ -889,10 +889,9 @@ static CYTHON_INLINE PyObject* __Pyx_PyInt_FromDouble(double value) {
         func_type value = func_value;                                     \
         if (sizeof(target_type) < sizeof(func_type)) {                    \
             if (unlikely(value != (func_type) (target_type) value)) {     \
-                func_type zero = 0;                                       \
                 if (exc && unlikely(value == (func_type)-1 && PyErr_Occurred()))  \
                     return (target_type) -1;                              \
-                if (is_unsigned && unlikely(value < zero))                \
+                if (is_unsigned && unlikely(value < (func_type)0))        \
```

Comment:
This change broke the build since it now generates tons of warnings with gcc. Note that other, similar cases in the same file use `neg_one = -1` and `value <= neg_one`, for similar reasons. That's the way to go, I think.

Reply:
This was not really related to the content of this PR anyways, so I will create a more appropriate fix in another PR